src/EnergyPlus/DesiccantDehumidifiers.cc

```diff
@@ -3335,144 +3335,146 @@ namespace DesiccantDehumidifiers {
 
         RegenCoilActual = 0.0;
         if (RegenCoilLoad > SmallLoad) {
-            {
-                auto const SELECT_CASE_var(DesicDehum(DesicDehumNum).RegenCoilType_Num);
-                if ((SELECT_CASE_var == Coil_HeatingGasOrOtherFuel) || (SELECT_CASE_var == Coil_HeatingElectric)) {
-                    SimulateHeatingCoilComponents(state,
-                                                  DesicDehum(DesicDehumNum).RegenCoilName,
-                                                  FirstHVACIteration,
-                                                  RegenCoilLoad,
-                                                  DesicDehum(DesicDehumNum).RegenCoilIndex,
-                                                  RegenCoilActual);
-                } else if (SELECT_CASE_var == Coil_HeatingWater) {
-                    MaxHotWaterFlow = DesicDehum(DesicDehumNum).MaxCoilFluidFlow;
-                    SetComponentFlowRate(state,
-                                         MaxHotWaterFlow,
-                                         DesicDehum(DesicDehumNum).CoilControlNode,
-                                         DesicDehum(DesicDehumNum).CoilOutletNode,
-                                         DesicDehum(DesicDehumNum).plantLoc);
-                    RegenCoilActual = RegenCoilLoad;
-                    // simulate the regenerator hot water heating coil
-                    SimulateWaterCoilComponents(state,
-                                                DesicDehum(DesicDehumNum).RegenCoilName,
-                                                FirstHVACIteration,
-                                                DesicDehum(DesicDehumNum).RegenCoilIndex,
-                                                RegenCoilActual);
-
-                    if (RegenCoilActual > (RegenCoilLoad + SmallLoad)) {
-                        // control water flow to obtain output matching RegenCoilLoad
-                        SolFlag = 0;
-                        MinWaterFlow = 0.0;
-                        std::array<Real64, 3> Par;
-                        Par[0] = double(DesicDehumNum);
-                        if (FirstHVACIteration) {
-                            Par[1] = 1.0;
-                        } else {
-                            Par[1] = 0.0;
+            switch (DesicDehum(DesicDehumNum).RegenCoilType_Num) {
```

Comment:
`auto & desicDehum = state.dataDesicDehum->DesicDehum(DesicDehumNum);` is a much more useful shortcut than `auto & DesicDehum = state.dataDesicDehum->DesicDehum;`. In general, don't have shortcut references to arrays, use them for individual elements.